### PR TITLE
Service knows whether it's in autofork mode or not

### PR DESCRIPTION
**Pull request by maxtaco**
This way we can be smarter about restarting it

Addresses maxtaco/CORE-2102

### CODE REVIEW

**Comment by maxtaco** [comment]
C/I failures are fixed in #1336

File: go/client/cmd_ctl_restart.go [hunk 1]
```diff
@@ -47,7 +56,7 @@ func (s *CmdCtlRestart) Run() error {
 	G.Log.Info("Delaying for shutdown...")
```
Comment (patrickxb) [inline go/client/cmd_ctl_restart.go]: s.G().Log

File: go/client/fork_server.go [hunk 2]
```diff
@@ -27,11 +27,16 @@ func GetExtraFlags() []cli.Flag {
 	}
 }
 
+// AutoForkServer just forks the server and sets the autoFork flag to true
+func AutoForkServer(cl libkb.CommandLine, g *libkb.GlobalContext) error {
+	return ForkServer(cl, g, true /* isAutFork */)
```
Comment (patrickxb) [inline go/client/fork_server.go]: isAutoFork (typo)

**Comment by patrickxb** [comment]
LGTM, minor comments

**Comment by maxtaco** [comment]
Thanks, I'll fix these in #1336

**Pull request merged**